android/src/com/frostwire/android/gui/activities/ToggleAllSearchEnginesPreference2.java

```diff
@@ -152,32 +144,15 @@ private boolean areAllEnginesChecked() {
         return true;
     }
 
-    private boolean isOneEngineNotChecked() {
-        int count = 0;
-        if (searchEnginePreferences != null) {
-            for (CheckBoxPreference preference : searchEnginePreferences.keySet()) {
-                if (preference != null) {
-                    if (!preference.isChecked()) {
-                        count++;
-                    }
-                }
-            }
-            if (count == 1) {
-                return true;
-            }
-        }
-        return false;
-    }
-
-    private boolean isAtLeastOneOtherEngineChecked(CheckBoxPreference checkBoxPreference) {
+    private boolean isTheOnlyCheckedEngine(CheckBoxPreference checkBoxPreference) {
         for (CheckBoxPreference preference : searchEnginePreferences.keySet()) {
             if (preference != null && preference != checkBoxPreference) {
                 if (preference.isChecked()) {
-                    return true;
+                    return false;
                 }
             }
         }
-        return false;
+        return true;
```

Comment:
You are right. This code will "break" if we have nothing checked (will not happen, but still...) we would not allow checking the preference. I added a check.